examples/onnx/pytorch/translation/bart_onnx/generation_onnx.py

```diff
@@ -0,0 +1,675 @@
+
+
+class DecoderForONNX(torch.nn.Module):
+    def __init__(self, decoder):
+        super().__init__()
+        self.decoder = decoder
+
+    def forward(self, input_ids, encoder_state, attention_mask, past = None):
+        all_results = None
+        if past is not None:
+            all_results = list_to_tuple(past)
+            input_ids = input_ids[:, -1:]
+
+        last_hidden_state, past_key_values = self.decoder(
+            input_ids=input_ids,
+            encoder_hidden_states=encoder_state,
+            encoder_attention_mask=attention_mask,
+            past_key_values=all_results,
+            return_dict=False,
+        )
+
+        past_values = []
+        for past in past_key_values:
+            past_values = past_values + list(past)
+        return last_hidden_state, past_values
+
+def create_traced_encoder(encoder, input_ids, attention_mask):
+    encoder_c = copy.deepcopy(encoder)
+    encoder_for_onnx = EncoderForONNX(encoder_c)
+
+    # return torch.jit.trace(encoder, (input_ids, attention_mask))
+    return torch.jit.trace(encoder_for_onnx, (input_ids, attention_mask))
+
+def create_traced_decoder(decoder, input_ids, encoder_state, attention_mask, past = None):
+    decoder_c = copy.deepcopy(decoder)
+    decoder_for_onnx = DecoderForONNX(decoder_c)
+    past_values = flatten_list(past)
+
+    # Do this twice so we got 2 different decoders for further work.
+    if past_values is None or len(past_values) == 0:
+        return torch.jit.trace(decoder_for_onnx, (input_ids, encoder_state, attention_mask))
+    else:
+        return torch.jit.trace(decoder_for_onnx, (input_ids, encoder_state, attention_mask, past_values))
+
+class BartConfigTS(BartConfig, torch.nn.Module):
+    def init_module(self):
+        torch.nn.Module.__init__(self)
+
+class MinLengthLogitsProcessorTS(torch.nn.Module):
+    r"""
+    :class:`transformers.LogitsProcessor` enforcing a min-length by setting EOS probability to 0.
+
+    Args:
+        min_length (:obj:`int`):
+            The minimum length below which the score of :obj:`eos_token_id` is set to :obj:`-float("Inf")`.
+        eos_token_id (:obj:`int`):
+            The id of the `end-of-sequence` token.
+    """
+
+    def __init__(self, min_length: int, eos_token_id: int):
+        super().__init__()
+
+        if not isinstance(min_length, int) or min_length < 0:
+            raise ValueError(f"`min_length` has to be a positive integer, but is {min_length}")
+
+        if not isinstance(eos_token_id, int) or eos_token_id < 0:
+            raise ValueError(f"`eos_token_id` has to be a positive integer, but is {eos_token_id}")
+
+        self.min_length = min_length
+        self.eos_token_id = eos_token_id
+
+    def forward(self, input_ids, scores) -> torch.Tensor:
+        cur_len = input_ids.shape[-1]
+        if cur_len < self.min_length:
+            scores[:, self.eos_token_id] = -float("inf")
+        return scores
+
+class BARTGenerator(torch.nn.Module, GenerationMixin):
```

Comment:
Think I'm fine with overwriting the methods of `generate()` for the sake of this example